### PR TITLE
refactor(templates): refine required dependencies

### PR DESCRIPTION
**Pull request by emmenko**
Testing out pnpm on a starter template yielded the following changes to ensure the dependencies resolve correctly.

### CODE REVIEW

File: application-templates/starter-typescript/src/components/channel-details/channel-details.spec.tsx [hunk 1]
```diff
@@ -8,16 +8,13 @@ import {
   mapResourceAccessToAppliedPermissions,
   type TRenderAppWithReduxOptions,
 } from '@commercetools-frontend/application-shell/test-utils';
-import { reportErrorToSentry } from '@commercetools-frontend/sentry';
```
Comment (emmenko) [inline application-templates/starter-typescript/src/components/channel-details/channel-details.spec.tsx:11]: We use sentry internally (not sure if a customer uses that), no need to have it included by default in the template then.

File: application-templates/starter-typescript/.gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 node_modules
+public
```
Comment (kark) [inline application-templates/starter-typescript/.gitignore:2]: Perhaps we could do the same in the `@commercetools-applications/merchant-center-template-starter`

**Review by kark — APPROVED** [review]
Thanks 💯

**Review by ddouglasz — APPROVED** [review]
👏

**Review by CarlosCortizasCT — APPROVED** [review]

File: application-templates/starter-typescript/.gitignore (hunk 2 above)
Comment (emmenko) [reply application-templates/starter-typescript/.gitignore:2]: Ah yes, thanks for noticing!

**Pull request merged**